lib/config.ml

```diff
@@ -170,13 +170,14 @@ let non_overlapping cs =
   in
   let rec check = function
     | []    -> ()
-    | s::ss -> if not (List.for_all
-                         (fun ss' -> Domain_name.Set.is_empty (Domain_name.Set.inter s ss'))
-                         ss)
-               then
-                 invalid_arg "overlapping names in certificates"
-               else
-                 check ss
+    | s::ss ->
+      if not (List.for_all (fun ss' ->
+          X509.Certificate.Host_set.is_empty (X509.Certificate.Host_set.inter s ss'))
+          ss)
+      then
+        invalid_arg "overlapping names in certificates"
+      else
+        check ss
```

Comment:
Not that my suggestion below is particularly better, but a recursive function here doesn't seem necessary?

```ocaml
(* module M = X509.Certificate.Host_set *)
let check ss =
  let ss_count, unique =
    (* count each ss element, and add it to a union of all ss elements.*)
    List.fold_left (fun (count,acc) next ->
      count + M.cardinal next , M.union acc next)
      (0, M.empty) ss
  in
  if ss_count <> M.cardinal unique (* if not, duplicates in ss were merged in the union *)
    then invalid_arg "overlapping names in certificates"
in
check namessets
```

Reply:
if you feel strongly about that, please submit a separate PR with your code. I don't find either version particularly more readable -- and will stick to the smaller diff